node/directives/validateUserAccess.ts

```diff
@@ -0,0 +1,122 @@
+import { AuthenticationError, ForbiddenError } from '@vtex/api'
+import type { GraphQLField } from 'graphql'
+import { defaultFieldResolver } from 'graphql'
+import { SchemaDirectiveVisitor } from 'graphql-tools'
+
+import type { AuthAuditTokenMetrics } from '../metrics/auth'
+import sendAuthMetric, { AuthMetric } from '../metrics/auth'
+import {
+  validateAdminToken,
+  validateApiToken,
+  validateStoreToken,
+} from './helper'
+
+export class ValidateUserAccess extends SchemaDirectiveVisitor {
```

Comment:
can we name it `ValidateStoreUserAccess` for clarity? (as the other directive will probably be called `ValidateAdminUserAccess`

Reply:
Done